magniv/core/task.py

```diff
@@ -78,15 +81,36 @@ def _is_valid_key(self, key) -> bool:
         """
         return bool(re.match(Task.KEY_PATTERN, key))
 
+    def _make_valid_resources(self, resources) -> Dict[str, str]:
+        """
+        Takes in a resources dictionary with generic parameters and converts it to the appropriate syntax
+        for our infrastructure
+        #TODO: enforce reasonable limits for quantity of resources
+
+        :param key: The generic resources dict from the task decorator
+        :return: A dict with correct syntax
+        """
+        clean_dict = {}
+
+        if resources is None:
+            return resources
+        if "cpu" in resources.keys():
+            clean_dict["limit_cpu"] = resources["cpu"]
+        if "memory" in resources.keys():
+            clean_dict["limit_memory"] = resources["memory"]
```

Comment:
do we want to consider erroring instead of letting bad params fall off?

Reply:
Yes let's not fail silently incase someone mis-spells "memory" or something like that